### PR TITLE
Avoid duplicating the "Filters" section

### LINKED ISSUE #79630: Editor : Right-clicking 'Filter Nodes' field duplicates 'Filters' section in dropdown
### Godot version

4.1.stable

### System information

Windows 11 (x64), Intel(R) Core(TM) i5-9400 CPU, Vulkan,  NVIDIA GeForce GTX 1660 Ti driver version 528.24

### Issue description

Every time you right click the Filter Nodes field, it duplicates the 'Filters' section in the dropdown.

https://github.com/godotengine/godot/assets/104696964/b8d867c9-ba31-47c5-91c8-23cfa02f1c89

### Steps to reproduce

Rightclick the Filter Nodes field.

### Minimal reproduction project

N/A

### LINKED ISSUE #79663: Scene tree filter menu icon doesn't get updated on theme switch
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Debian GNU/Linux trixie/sid trixie - Vulkan (Compatibility) - Mesa Intel(R) HD Graphics 520 (SKL GT2) () - Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz (4 Threads)

### Issue description

After right clicking in thet "Filter Nodes" menu in the scene tree dock, the sub-menu is populated with a "Search" icon.

If I later switch themes in the same session, this icon is never updated to match the new theme.

image

### Steps to reproduce

1. Right click in the "Filter Nodes" menu in scene tree dock
2. Switch themes
3. Right click again in the "Filter Nodes" menu

Expected result: Icon has updated to match the theme I am currently using
Actual result: Icon is still using the theme I launched Godot with

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by garychia**
Fixes #79630
*Bugsquad edit:* Fixes #79663 Supersedes #79664

### CODE REVIEW

**Comment by jonhermansen** [comment]
I found a related bug while testing this change, but actually it has existed for some time.

I opened a new PR that includes both fixes but if that's not right, let me know. I would be happy to raise a separate bug and create a PR including just the icon fix.

**Comment by KoBeWi** [comment]
I think the original code makes little sense. Why is this method called on every popup if the menu is supposed to always be the same? It should only be initialized once, right after `filter` is created.

**Comment by garychia** [comment]
I agree. This seems to be a better solution.

**Comment by garychia** [comment]
I tried to initialize the menu right after `filter` is created, but the images on the menu were broken:
image
The original code seems to have taken this into consideration.

**Comment by KoBeWi** [comment]
The icons should be assigned in NOTIFICATION_THEME_CHANGED.

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -1386,6 +1386,11 @@ void SceneTreeDock::_notification(int p_what) {
 
 			filter->set_right_icon(get_theme_icon(SNAME("Search"), SNAME("EditorIcons")));
 
+			PopupMenu *p_menu = filter->get_menu();
```
Comment (KoBeWi) [inline editor/scene_tree_dock.cpp:1389]:
```suggestion
			PopupMenu *filter_menu = filter->get_menu();
```
`p_` prefix should be used only for **p**arameters.

**Comment by KoBeWi** [comment]
You should remove `get_theme_icon()` calls from `_append_filter_options_to()`.

File: editor/scene_tree_dock.cpp [hunk 2]
```diff
@@ -1386,6 +1386,11 @@ void SceneTreeDock::_notification(int p_what) {
 
 			filter->set_right_icon(get_theme_icon(SNAME("Search"), SNAME("EditorIcons")));
 
+			PopupMenu *filter_menu = filter->get_menu();
+			filter_menu->set_item_icon(filter_menu->get_item_idx_from_text(TTR("Filters")), get_theme_icon(SNAME("Search"), SNAME("EditorIcons")));
```
Comment (KoBeWi) [inline editor/scene_tree_dock.cpp:1390]: I wonder if there is a better way to get this item than using a text.

**Review by KoBeWi — APPROVED** [review]

**Comment by YuriSizov** [comment]
Looks like we forgot to say "thanks", so here's Thanks!
Cherry-picked for 4.1.2.

**Pull request merged**